docs/rules/valid-jsdoc.md

```diff
@@ -1,296 +1,358 @@
+    this.num1 = num1;
+    this.num2 = num2;
+}
 
-// use of @override make @param and @returns optional
+// class constructor allows missing @returns tag
 /**
- * A description
- * @override
+ * Represents a sum.
  */
-function foo(a) {
-    return a;
+class Sum {
+    /**
+     * @param {number} num1 The first number.
+     * @param {number} num2 The second number.
+     */
+    constructor(num1, num2) {
+        this.num1 = num1;
+        this.num2 = num2;
+    }
 }
 
-// @returns is not required for a constructor
-class Foo {
+// @abstract tag allows @returns tag without `return` statement
+class Widget {
     /**
-    *
-    * @param {int} num1 The first number.
+    * When the state changes, does it affect the rendered appearance?
+    * @abstract
+    * @param {Object} state The new state of the widget.
+    * @returns {boolean} Is current appearance inconsistent with new state?
     */
-    constructor(num1) {
-        this.num1 = num1;
+    mustRender (state) {
+        throw new Error("Widget subclass did not implement mustRender");
     }
 }
 
-// @returns allowed without return if used with @abstract
-class Foo {
+// @override tag allows missing @param and @returns tags
+class WonderfulWidget extends Widget {
     /**
-     * @abstract
-     * @return {Number} num
+     * @override
      */
-    abstractMethod () {
-        throw new Error('Not implemented');
+    mustRender (state) {
+        return state !== this.state; // shallow comparison
     }
 }
-
 ```
 
 ## Options
 
-### prefer
-
-JSDoc offers a lot of tags with overlapping meaning. For example, both `@return` and `@returns` are acceptable for specifying the return value of a function. However, you may want to enforce a certain tag be used instead of others. You can specify your preferences regarding tag substitution by providing a mapping called `prefer` in the rule configuration. For example, to specify that `@returns` should be used instead of `@return`, you can use the following configuration:
-
-```json
-"valid-jsdoc": ["error", {
-    "prefer": {
-        "return": "returns"
-    }
-}]
-```
+This rule has an object option:
 
-With this configuration, ESLint will warn when it finds `@return` and recommend to replace it with `@returns`.
+* `"prefer"` enforces consistent documentation tags specified by an object whose properties mean **disallow** the key and **allow** the value (for example, `"return": "returns"` means disallow `@return` and allow `@returns`)
+* `"preferType"` enforces consistent type strings specified by an object whose properties mean **disallow** the key and **allow** the value (for example, `"object": "Object"` means disallow `object` and allow `Object`)
+* `"requireReturn": true` (default) requires a return tag **even if** the function or method does not have a `return` statement (but it does not apply to constructor functions)
+* `"requireReturn": false` requires a return tag **if and only if** the function or method has a `return` statement
```

Comment:
Aha, I see what you mean.

I tried first to combine into one sentence, but in rare situations like this where `true` and `false` are not exactly opposites, what do you think about second-level bulleted list?

Whatever we do, there might be a few other rule docs where I need to correct this anti-pattern.